src/Amazon.Common.DotNetCli.Tools/DockerCLIWrapper.cs

```diff
@@ -163,18 +164,52 @@ public int Push(string targetTagName)
         /// <returns></returns>
         public int Run(string imageId, string containerName, string commandToRun = "")
         {
-            var containerNameArg = $"--name {containerName}";
+            var argumentList = new List<string>()
+            {
+                "run",
+                "--name",
+                containerName,
 
-            var autoDeleteArg = "--rm"; // Automatically remove the container once it's done running
+                // Automatically remove the container once it's done running
+                "--rm",
 
-            // This allows the container access to the working directory in a virtual mapped path located at /tmp/source
-            // That means that when the container is finished running, anything it leaves in /tmp/source (e.g. the binaries),
-            // will just exist in the working directory
-            var mountArg = $"--volume \"{this._workingDirectory}\":{WorkingDirectoryMountLocation}";
+                // This allows the container access to the working directory in a virtual mapped path located at /tmp/source
+                // That means that when the container is finished running, anything it leaves in /tmp/source (e.g. the binaries),
+                // will just exist in the working directory
+                "--volume",
+                $"\"{this._workingDirectory}\":{WorkingDirectoryMountLocation}",
 
-            var imageArg = $"-i {imageId}";
+                "-i"
+            };
+            
+            // For Linux or MacOS, running a .NET image as non-root requires some special
+            // handling.  We have to pass the user's UID and GID, but we also have to specify
+            // the location of where .NET and Nuget put their files, because by default, they
+            // go to the /.dotnet and /.local, which a non-root user can't write to in a container
+            if (PosixUserHelper.IsRunningInPosix)
```

Comment:
If the user id is `0` then I think we should skip this part since we are already running at root.

Reply:
Ok, changing this to skip if PosixUserHelper isn't able to get the user or group, or if the user and group are both zero (root):

```
                var posixUser = PosixUserHelper.GetEffectiveUser(_logger);
                if (posixUser?.UserID > 0 posixUser?.GroupID > 0)
                {
```